libs/dfi/src/avrobin_serializer.c

```diff
@@ -692,7 +692,7 @@ static int avrobinSerializer_writeSequence(dyn_type *type, void *loc, FILE *stre
 
 static int avrobinSerializer_writeEnum(dyn_type *type, void *loc, FILE *stream) {
     char enum_value_str[16];
-    if (sprintf(enum_value_str, "%d", *(int32_t*)loc) < 0) {
+    if (snprintf(enum_value_str, sizeof(enum_value_str), "%d", *(int32_t*)loc) >= sizeof(enum_value_str)) {
```

Comment:
According to `man snprintf`, if an output error is encountered, a negative value is returned.